Fix `cell_diffusion_timescale` with zero tracers

ali-ramadhan opened the pull request.

glwagner commented:
Might be better to use dispatch here (on the case where the kappa-tuple has length zero) for performance, clarity, and julianism!

glwagner commented on src/TurbulenceClosures/turbulence_closure_diagnostics.jl:
```diff
@@ -8,26 +8,38 @@ cell_diffusion_timescale(model) = cell_diffusion_timescale(model.closure, model.
-    return min(Δz^2 / closure.νv, Δh^2 / closure.νh,
-               Δz^2 / max_κv, Δh^2 / max_κh)
+    if length(closure.κ) == 0
+        return min(Δz^2 / closure.νv, Δh^2 / closure.νh)
```
Could define a function `cell_viscous_time_scale` that returns this

ali-ramadhan commented:
I thought of defining `cell_viscous_timescale` and `cell_tracer_diffusion_timescale` so that `cell_diffusion_timescale` just returns
```julia
min(cell_viscous_timescale, cell_tracer_diffusion_timescale)
```
but then `cell_tracer_diffusion_timescale` would have to return `Inf` instead of `nothing` and it seemed a bit messy with lots of one-liner functions and duplicated dispatch signatures so I just refactored `cell_diffusion_timescale` to just dispatch on whether `closure.κ` is an empty named tuple or not.

I added tests to make sure `cell_diffusion_timescale` will work for all closures when `buoyancy=nothing, tracers=nothing`.
